### PR TITLE
Update CONTRIBUTING to reflect latest changes in build and deployment

### PR DESCRIPTION
**Pull request by lucferbux**
Update the CONTRIBUTING section to reflect the latest changes in building and deploying the dashboard into a cluster.

### CODE REVIEW

**Review by andrewballantyne — COMMENTED** [review]
I typically just deploy the image directly onto the cluster by modifying the image field -- I guess since you remove it from the KfDef, this will manage it all via the `makefile`? 

We'll have to update our steps when the KfDefs get swapped out for the new operator & CR.

**Comment by lucferbux** [comment]
> I typically just deploy the image directly onto the cluster by modifying the image field -- I guess since you remove it from the KfDef, this will manage it all via the `makefile`?
> 
> We'll have to update our steps when the KfDefs get swapped out for the new operator & CR.

You mean modifying the image field in the deployment right?
This method allows you to also deploy new resources if the `manifest` folder is changed, you can achieve the same result changing the repo reference in the `kfdef` spec to point at some PR. But with this method, you can even test changes without raising a PR.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -201,10 +178,12 @@ make build
...
 
 ```bash
-npm run make:build
+npm run make:deploy
 ```
 
-### Pushing the image
+in the root of this repository. By default, we use [podman](https://podman.io/) as the default container tool, but you can change it by setting the `CONTAINER_TOOL` environment variable to `docker`.
+
+After bulding the image, you need to push it to a container registry accessible by your cluster. You can do that by running
```
Comment (manaswinidas) [inline CONTRIBUTING.md:186]: Minor tweak:

Typo: `bulding` -> `building`

**Review by manaswinidas — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -201,10 +178,12 @@ make build
 or
 
 ```bash
-npm run make:build
+npm run make:deploy
```
Comment (pnaik1) [inline CONTRIBUTING.md:181]: @lucferbux , shouldn't it be `npm run make:build` ?

File: CONTRIBUTING.md (hunk 1 above)
Comment (lucferbux) [reply CONTRIBUTING.md:186]: Thanks!

File: CONTRIBUTING.md (hunk 2 above)
Comment (lucferbux) [reply CONTRIBUTING.md:181]: @pnaik1 Yes that's right, thanks a lot!

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -230,6 +215,4 @@ or
 npm run make:deploy
 ```
 
-### Pull Request Images
-
-All pull requests will have an associated `pr-<PULL REQUEST NUMBER>` image built and pushed to [quay.io](https://quay.io/repository/opendatahub/odh-dashboard) for use in testing and verifying code changes as part of the PR code review.  Any updates to the PR code will automatically trigger a new PR image build, replacing the previous hash that was referenced by `pr-<PULL REQUEST NUMBER>`.
+you will deploy all the resources located in the `manifest` folder alongside the image you selected in the previous step.
```
Comment (manaswinidas) [inline CONTRIBUTING.md:218]: Minor typo:

`manifests` folder 

Sorry I missed this the first time.

**Review by manaswinidas — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 3 above)
Comment (lucferbux) [reply CONTRIBUTING.md:218]: thanks a lot!

**Review by manaswinidas — APPROVED** [review]
/lgtm

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -213,12 +192,18 @@ make push
 or
 
 ```bash
-npm run make:push
+npm make:push
```
Comment (pnaik1) [inline CONTRIBUTING.md:195]: @lucferbux, A last one  can you change this to `npm run make:push`

**Comment by lucferbux** [comment]
@pnaik1 can you re-review it please?

**Comment by pnaik1** [comment]
@lucferbux  Can you take a look at the test , its failing

**Comment by lucferbux** [comment]
@pnaik1 don't worry, this is due to an issue in the cache, let me fix it

**Pull request merged**